### PR TITLE
A.22 assignment

### PR DESCRIPTION
**Pull request by ravin-imm**

### CODE REVIEW

File: PSICover/PSICover.cs [hunk 1]
```diff
@@ -163,11 +166,20 @@ class Analyzer {
          var code = File.ReadAllLines (file);
          for (int i = 0; i < code.Length; i++)
             code[i] = code[i].Replace ('<', '\u00ab').Replace ('>', '\u00bb');
+         int nHits = 0;
```
Comment (sdinesh-90) [inline PSICover/PSICover.cs]: Review from Arvind: Minor style guideline. I generally use lowercase c as a prefix to indicate a count. So this variable would be called `cHits`.

File: PSICover/PSICover.cs [hunk 2]
```diff
@@ -163,11 +166,20 @@ class Analyzer {
...
             code[i] = code[i].Replace ('<', '\u00ab').Replace ('>', '\u00bb');
+         int nHits = 0;
          foreach (var block in blocks) {
             bool hit = hits[block.Id] > 0;
-            string tag = $"<span class=\"{(hit ? "hit" : "unhit")}\">";
-            code[block.ELine] = code[block.ELine].Insert (block.ECol, "</span>");
-            code[block.SLine] = code[block.SLine].Insert (block.SCol, tag);
+            if (hit) nHits++;
+            string tag = $"<span {(hit ? $"class=\"hit\" title=\"Hits: {hits[block.Id]}\"" : "class=\"unhit\"")}>";
+            if (block.ELine != block.SLine) {
```
Comment (sdinesh-90) [inline PSICover/PSICover.cs]: If condition is not needed.

**Review by sdinesh-90 — CHANGES_REQUESTED** [review]

File: PSICover/PSICover.cs (hunk 2 above)
Comment (ravin-imm) [reply PSICover/PSICover.cs]: @sdinesh-90 : I think we need to consider the start and the end columns when it is a single line block. Can you please elaborate on the change required?

Comment (sdinesh-90) [reply PSICover/PSICover.cs]: For every line of code, we need to figure out start and end column. So, based on first or last line, we could set the column index appropriately.
Even though this uses two if condition for every loop, it seems to be precise.

**Review by sdinesh-90 — APPROVED** [review]

**Comment by sdinesh-90** [comment]
Looks fine

**Pull request closed without merging**